java/java.source.base/src/org/netbeans/modules/java/source/parsing/OutputFileManager.java

```diff
@@ -116,22 +114,28 @@ public OutputFileManager(
             File activeRoot = getClassFolderForSource(l, sibling, baseName);
             if (activeRoot == null) {
                 activeRoot = getClassFolderForApt(l, sibling, baseName);
+            }
+            if (activeRoot == null && siblings.hasSibling()) {
```

Comment:
This is the meat of the fix - the above attempts to compute the target class folder might have failed, because, if sibling is null, it will try to find the original source root based on "baseName" - but if the newly generated classfile's name does not correspond to any source file (which is valid, of course), this search fails. So adding a new layer, using the "siblings" utility.